lib/descriptors/MessageDescriptor.js

```diff
@@ -22,6 +22,8 @@ function MessageDescriptor(name) {
   this._enums = {}
   this._fields = {}
   this._fieldNames = {}
+  this._oneofs = {}
```

Comment:
Yeah, I went back and forth on this. Now that there's an assertion in addOneof, I think it makes sense to convert back to an array.